src/plots/cartesian/axes.js

```diff
@@ -82,21 +82,92 @@ axes.coerceRef = function(containerIn, containerOut, gd, attr, dflt, extraOption
     var refAttr = attr + 'ref';
     var attrDef = {};
 
-    if(!dflt) dflt = axlist[0] || extraOption;
+    if(!dflt) {
+        dflt = axlist[0] || (
+        ((extraOption === undefined) || (typeof(extraOption) === 'string')) ?
+            extraOption :
+            extraOption[0]
+        );
+    }
+    // so in the end, if dflt, axlist[], and extraOption were undefined, they
+    // are still undefined
     if(!extraOption) extraOption = dflt;
 
     // data-ref annotations are not supported in gl2d yet
 
+    var extraOptionAsList = extraOption;
+    if(extraOption) {
+        if(typeof(extraOption) === 'string') {
+            extraOptionAsList = [extraOption];
+        }
+    }
+
     attrDef[refAttr] = {
         valType: 'enumerated',
-        values: axlist.concat(extraOption ? [extraOption] : []),
+        values: axlist.concat(extraOptionAsList ? extraOptionAsList : []),
         dflt: dflt
     };
 
     // xref, yref
     return Lib.coerce(containerIn, containerOut, attrDef, refAttr);
 };
 
+/*
+ * An axis reference (e.g., the contents at the 'xref' key of an object) might
+ * have extra information appended. Extract the axis reference only.
+ *
+ * ar: the axis reference string
+ *
+ */
+axes.extractAxisFromAxisRef = function(ar) {
```

Comment:
The terminology I think we use elsewhere is:
- "axis id" - a string like `'x'`, `'y2'` used to refer to a specific axis
- "axis name" - a string like `'xaxis'`, `'yaxis2'` that's also the container name in `layout`
- "axis" - the `_fullLayout` object

"axis reference" we perhaps haven't settled on a meaning yet, but I'd suggest using it to mean anything that can be set as the value of an `xref` or `yref` attribute. Then this utility might belong in [cartesian/axis_ids.js](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/axis_ids.js) as something like `exports.ref2id` to match the other function names there.

Also you're always feeding it the output of a `coerceRef` call, so you know the input is always going to be a valid axis reference - so you should be able to simplify this to something like "if the first character is an x/y/z, return everything before the first space; otherwise return false" (I feel like `false` would be more useful in the `'paper'` case than still returning `'paper'`, would you agree?)

Reply:
:+1: